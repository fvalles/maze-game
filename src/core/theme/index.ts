export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
