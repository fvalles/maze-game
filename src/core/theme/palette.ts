export const palette = {
  // color names are based on https://colornamer.netlify.app/
  artfulRed: '#B20600',
  crystalCut: '#F9F2ED',
  blackMarket: '#212240',
  janitor: '#2666CF',
  white: '#FFFFFF',
};
