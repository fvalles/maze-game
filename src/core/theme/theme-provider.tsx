import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
