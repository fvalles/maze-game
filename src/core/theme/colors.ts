import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  buttonBackground: palette.artfulRed,
  modalBackground: palette.crystalCut,
  restartButtonBackground: palette.janitor,
  text: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
