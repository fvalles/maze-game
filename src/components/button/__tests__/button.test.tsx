import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Button } from '..';

describe('button component', () => {
  it('should render with title, onClick function and default background and text colors', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Button
            onClick={jest.fn()}
            testId="some-test-id"
            title="Some Title"
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('background-color', '#B20600');
    expect(result).toHaveStyleRule('border-radius', '4px');
    expect(result).toHaveStyleRule('padding-bottom', '16px');
    expect(result).toHaveStyleRule('padding-top', '16px');
    expect(result).toHaveStyleRule('width', '200px');

    const buttonText = result.children[0];
    expect(buttonText).toHaveStyleRule('color', '#FFFFFF');
    expect(result).toMatchSnapshot();
  });

  it('should render with title, onClick function and custom background and text', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Button
            background="white"
            color="text"
            onClick={jest.fn()}
            testId="some-test-id"
            title="Some Title"
          />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('background-color', '#FFFFFF');
    expect(result).toHaveStyleRule('border-radius', '4px');
    expect(result).toHaveStyleRule('padding-bottom', '16px');
    expect(result).toHaveStyleRule('padding-top', '16px');
    expect(result).toHaveStyleRule('width', '200px');

    const buttonText = result.children[0];
    expect(buttonText).toHaveStyleRule('color', '#212240');
    expect(result).toMatchSnapshot();
  });
});
