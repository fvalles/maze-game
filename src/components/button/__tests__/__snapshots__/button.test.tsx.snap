// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render with title, onClick function and custom background and text 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 16px;
}

.c0 {
  background-color: #FFFFFF;
  border-radius: 4px;
  padding-bottom: 16px;
  padding-top: 16px;
  width: 200px;
}

<button
  className="c0"
  data-testid="some-test-id"
  onClick={[MockFunction]}
>
  <span
    className="c1"
    color="text"
  >
    Some Title
  </span>
</button>
`;

exports[`button component should render with title, onClick function and default background and text colors 1`] = `
.c1 {
  color: #FFFFFF;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 16px;
}

.c0 {
  background-color: #B20600;
  border-radius: 4px;
  padding-bottom: 16px;
  padding-top: 16px;
  width: 200px;
}

<button
  className="c0"
  data-testid="some-test-id"
  onClick={[MockFunction]}
>
  <span
    className="c1"
  >
    Some Title
  </span>
</button>
`;
